src/components/MainCard.jsx

```diff
@@ -0,0 +1,56 @@
+import { useContext, useState } from "react";
+import { BsArrowDownUp } from "react-icons/bs";
+import styles from "./MainCard.module.css";
+import buttonStyles from "./ToggleButton.module.css";
+import Input from "./Input";
+import ToggleButton from "./ToggleButton";
+import DatePicker from "./DatePicker";
+import { TripContext } from "./TripContext";
+
+export default function MainCard() {
+  const [origin, setOrigin] = useState(() => "");
+  const [destiny, setDestiny] = useState(() => "");
+  const [dateTime, setDateTime] = useState(() => undefined);
+  const [route, setRoute] = useState(() => "departure");
+  const [clicked, setClicked] = useState(false);
+
+  const { planJourney } = useContext(TripContext);
+
+  return (
+    <div className={`${styles.card} ${clicked && styles.transition}`}>
+      <div className={styles.cardContent}>
+        <Input placeholder="From:" onSelect={(station) => setOrigin(station)} />
```

Comment:
```suggestion
        <Input placeholder="From:" onSelect={setOrigin} />
```
It is just a suggestion, you can clean a bit the syntax just passing the function as parameter, since it will match the method signature.

Reply:
In other parts of your code you have similar cases, so if you decide to follow my tip, you can do it everywhere.

Reply:
Seems interesting, I didn´t know that passing the state as a parameter would automatically trigger the props passing the value to the state.